mscore/musescore.cpp

```diff
@@ -2001,7 +2001,7 @@ MuseScore::MuseScore()
             const bool loadSuccess = _loginManager->load();
 
             if (cliSaveOnline && !loadSuccess) {
-                  qFatal(qUtf8Printable(tr("No login credentials stored. Please sign in via the GUI.")));
+                  qFatal("No login credentials stored. Please sign in via the GUI.");
```

Comment:
Veto. This way that text is not translatable

Reply:
No other qFatal is translatable. (I don't think that qFatal, qWarning, and other developer messages need translation.)

Reply:
IMHO they do, all of them

Reply:
Not the qDebug() ones though, not sure about qWarning().

Reply:
If at that place the translations engine is available, which for example it is not at the place wherethe Usage message is printed

Reply:
At this place translations are avaibable though, as far as I can tell.

Reply:
How about `qFatal("%s", qUtf8Printable(tr("No login credentials stored. Please sign in via the GUI.")));`

Reply:
@orivej, does this work:

```c++
QString msg = tr("No login credentials stored. Please sign in via the GUI.");
qFatal(qUtf8Printable(msg));
```
Or
```c++
auto msg = qUtf8Printable(tr("No login credentials stored. Please sign in via the GUI."));
qFatal(msg);
```
?

Reply:
Or that, yes

Reply:
I meant that no other qFatal (or qWarning, or qDebug) uses `tr` and therefore is not translatable. Why does this specific qFatal have to be the only one with `tr`?

Reply:
Because the other `qFatal()`s should too

Reply:
Esp. those that are related to running MuseScore from the commandline (does `qFatal()` bring up a dialog in GUI mode?)

Reply:
`qFatal(qUtf8Printable(msg))` and `qFatal(msg)` do not pass `-Wformat-security`: it protects against the case where the translated msg contains `%s` or other special formatting directives.

`qFatal("%s", …)` works. However, qFatal should not be used for user error reporting (which seems to be the case here), it should be used for violated developer assumptions (like `abort()` in C) because it creates a core dump: https://doc.qt.io/qt-5/qtglobal.html#qFatal . This is also why it should not be translated: it is easier to grep the code for the error message rather than to correlate the translated error message to the original.

Reply:
Ok, so here we might better be using `fprintf(stderr, "...")` or `std::cerr << ...`

Reply:
@Jojo-Schmitz, what about `qWarning() << "..."` followed by `exit(EXIT_FAILURE)`?

Reply:
Would work too I guess. Not sure though whether either of these functions prefixes the (error) message with "Fatal:", "Warning:" or some such

Reply:
and whether and how it works on Windows commandline (but there I'm not sure about `cerr?  and the likes either)

Reply:
OK. I had to use `::exit` because `exit` resolves into `QCoreApplication::exit()` (exit from the main loop), which does nothing when the main loop is not running.